File: java/src/dev/csn/Aos.java
package dev.csn;

import java.util.ArrayList;
import java.util.Random;

public class Aos {
    public ArrayList<ListItem> items;
    public int size;

    Aos(int size) {
        items = new ArrayList<>(size);
        this.size = size;
        fill(size);
    }

    public void fill(int size) {
        var r = new Random();
        for (var i = 0; i < size; i++) {
            var item = new ListItem();
            item.itemId = r.nextInt();
            item.selected = r.nextInt(10) > 5;
            items.add(item);
        }
    }

    public int[] randomIds() {
        var lids = new int[size];
        var rids = new int[size];
        for (var i = 0; i < size; i++) {
            lids[i] = items.get(i).itemId;
        }
        var r = new Random();
        for (var i = 0; i < size; i++) {
            rids[i] = lids[r.nextInt(size)];
        }
        return rids;
    }

    public boolean selected(int id) {
        for (ListItem item : items) {
            if (item.itemId == id) {
                return item.selected;
            }
        }
        return false;
    }
}

File: kotlin/src/main/kotlin/Aos.kt
import kotlin.random.Random

class Aos(size_: Int) {
    val size: Int
    var items: ArrayList<ListItem>

    init {
        size = size_
        val itemsInit = Array(size) { i ->
            ListItem(Random.nextInt(), Random.nextBoolean(), "Item $i", "http://items.dev/$i")
        }
        items = ArrayList(size)
        items.addAll(itemsInit)
    }

    fun randomIds(): Array<Int> {
        val lids = Array(size) { i ->
            items[i].id
        }
        val rids = Array(size) {
            lids[Random.nextInt(0, size)]
        }

        return rids
    }

    fun selected(id: Int): Boolean {
        for (item in items) {
            if (item.id == id) {
                return item.selected
            }
        }
        return false
    }
}